admin_pages/transactions/Transactions_Admin_Page.core.php

```diff
@@ -752,23 +734,21 @@ protected function _transaction_details()
         }
         $this->_template_args['amount_due_class'] = '';
 
-        if ($this->_transaction->get('TXN_paid') == $this->_transaction->get('TXN_total')) {
+        if ((float) $this->_transaction->paid() === (float) $this->_transaction->total()) {
             // paid in full
             $this->_template_args['amount_due'] = false;
-        } elseif ($this->_transaction->get('TXN_paid') > $this->_transaction->get('TXN_total')) {
+        } elseif ($this->_transaction->paid() > $this->_transaction->total()) {
             // overpaid
             $this->_template_args['amount_due_class'] = 'txn-overview-no-payment-spn';
-        } elseif ($this->_transaction->get('TXN_total') > 0
-                  && $this->_transaction->get('TXN_paid') > 0
-        ) {
-            // monies owing
-            $this->_template_args['amount_due_class'] = 'txn-overview-part-payment-spn';
-        } elseif ($this->_transaction->get('TXN_total') > 0
-                  && $this->_transaction->get('TXN_paid') == 0
-        ) {
-            // no payments made yet
-            $this->_template_args['amount_due_class'] = 'txn-overview-no-payment-spn';
-        } elseif ($this->_transaction->get('TXN_total') == 0) {
+        } elseif ((float) $this->_transaction->total() > (float) 0) {
+            if ((float) $this->_transaction->paid() > (float) 0) {
+                // monies owing
+                $this->_template_args['amount_due_class'] = 'txn-overview-part-payment-spn';
+            } elseif ((float) $this->_transaction->paid() === (float) 0) {
```

Comment:
`EE_Transaction::paid` already returns a float, why cast again?

Reply:
missed that, will fix